pkg/agent/proxy/proxier.go

```diff
@@ -410,11 +464,18 @@ func (p *proxier) installServices() {
 
 		installedSvcPort, ok := p.serviceInstalledMap[svcPortName]
 		var pSvcInfo *types.ServiceInfo
-		var needRemoval, needUpdateService, needUpdateEndpoints bool
+		var needUpdateServiceExternalAddresses, needUpdateService, needUpdateEndpoints bool
```

Comment:
It may be possible to be more fine-grained, however, I think it's not very worth to save a few calls but introduce many special processing for some infrequent operations. Unless it can be done without adding many complexities, I feel it's unnecessary.